ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -320,16 +320,23 @@ def _switch_data(self):
         node_config = {}
         hostname = self.hostname
         node_type_config = get(self._hostvars, f"{self.node_type_key}", required=True)
-
-        if hostname in node_type_config.get('nodes', {}):
-            node_config = node_type_config['nodes'][hostname]
-        else:
-            for node_group in node_type_config.get('node_groups', {}):
-                if hostname in node_type_config['node_groups'][node_group].get('nodes', {}):
-                    node_config = node_type_config['node_groups'][node_group]['nodes'][hostname]
-                    switch_data['node_group'] = node_type_config['node_groups'][node_group]
-                    switch_data['group'] = node_group
-                    break
+        nodes = self._convert_dicts(node_type_config.get('nodes', []), 'name')
+
+        for node in nodes:
+            if hostname == node['name']:
+                node_config = node
+
+        if not node_config:
+            node_groups = self._convert_dicts(node_type_config.get('node_groups', []), 'group')
+            for node_group in node_groups:
+                nodes = self._convert_dicts(node_group.get('nodes', []), 'name')
+                node_group['nodes'] = nodes
+                for node in nodes:
+                    if hostname == node['name']:
+                        node_config = node
+                        switch_data['node_group'] = node_group
+                        switch_data['group'] = node_group['group']
+                        break
```

Comment:
Given that we reuse the pattern quite often of finding an entry of the list of dict based on a value for a given key, I am wondering if adding a function in utils would be useful @ClausHolbechArista ?

Reply:
I have a function like that in one of my PRs. Let's create a toolbox when we start the larger move to python